packages/core/orchestration/src/joiner/remote-joiner.ts

```diff
@@ -1203,8 +1207,21 @@ export class RemoteJoiner {
       queryObj,
     })
 
+    if (options?.throwIfKeyNotFound) {
+      if (!primaryKeyArg || !primaryKeyArg.value) {
```

Comment:
sugg: should we split this into 2 different error messages to make it more specific? i.e. `primary keys weren't found` and then `value for primary keys weren't found`

Reply:
yeah. we can do that